Godot editor build fails at 62% with disable_xr=yes

### Tested versions

-Reproducible in v4.6.dev.custom_build [0400b703e].

### System information

Linux Mint 22.2. HP laptop with Ryzen 5 5625U and integrated graphics.

### Issue description

Editor build crash at 62% with this error:

[ 62%] Compiling scene/gui/button.cpp ...
scene/main/viewport.cpp:1122:7: error: use of undeclared identifier 'use_xr'
 1122 |         if (!use_xr) {
      |              ^
[ 62%] Compiling scene/gui/center_container.cpp ...
1 error generated.
scons: *** [bin/obj/scene/main/viewport.linuxbsd.editor.x86_64.llvm.o] Error 1
scons: building terminated because of errors.

I've tested this some times and the error occours only with disable_xr=yes


### Steps to reproduce

Build godot editor from the latest main with disable_xr=yes.

custom.py
platform="linuxbsd"
tagrget="editor"
arch="x86_64"
use_llvm="yes" 
linker="lld"
deprecated="no"
prodution="yes"
module_camera_enabled = "no"
module_gltf_enabled = "no"
disable_xr="yes"
module_mobile_vr_enabled = "no"
module_openxr_enabled = "no"
module_webxr_enabled = "no"
builtin_embree="no" 
builtin_enet="no" 
builtin_freetype="no"
builtin_graphite="no"
builtin_harfbuzz="no"
builtin_libogg="no"
builtin_libpng="no" 
builtin_libtheora="no" 
builtin_libvorbis="no" 
builtin_libwebp="no" 
builtin_mbedtls="no"
builtin_miniupnpc="no" 
builtin_pcre2="no"
builtin_zlib="no" 
builtin_zstd="no"

### Minimal reproduction project (MRP)

There's no project since is an error generated building the engine.

--- a/scene/main/viewport.cpp
+++ b/scene/main/viewport.cpp
@@ -1118,7 +1118,7 @@ bool Viewport::_set_size(const Size2i &p_size, const Size2 &p_size_2d_override, 
 	stretch_transform = stretch_transform_new;
 	font_oversampling = new_font_oversampling;
 
-#ifndef _3D_DISABLED
+#ifndef XR_DISABLED
 	if (!use_xr) {
 #endif
 
@@ -1128,7 +1128,7 @@ bool Viewport::_set_size(const Size2i &p_size, const Size2 &p_size_2d_override, 
 			RS::get_singleton()->viewport_set_size(viewport, 0, 0);
 		}
 
-#ifndef _3D_DISABLED
+#ifndef XR_DISABLED
 	} // if (!use_xr)
 #endif
 

